test/server/proxy-test.js

```diff
@@ -795,5 +796,33 @@ describe('Proxy', function () {
                 done();
             });
         });
+
+        it('Should pass ECONNREFUSED error to session (GH-446)', function (done) {
+            freeport(function (error, port) {
```

Comment:
Tip: you can use awesome [endpoint-utils](https://github.com/inikulin/endpoint-utils) package that has promise-based functionality, so you will not need ugly callbacks :smile: